File: Backend/src/main/java/com/luckyseven/backend/domain/member/service/utill/MemberDetails.java
package com.luckyseven.backend.domain.member.service.utill;

import com.luckyseven.backend.domain.member.entity.Member;
import java.util.Collection;
import java.util.List;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

public class MemberDetails implements UserDetails {
  private final Long id;
  private final String password;
  private final String email;
  private final String nickname;
  private final Collection<? extends GrantedAuthority> authorities;

  public MemberDetails(Member member) {
    this.id          = member.getId();
    this.email       = member.getEmail();
    this.nickname     = member.getNickname();
    this.password = member.getPassword();
    this.authorities = List.of(new SimpleGrantedAuthority("ROLE_USER"));
  }

  public MemberDetails(Long id, String password, String email, String nickname) {
    this.id = id;
    this.password = password;
    this.email = email;
    this.nickname = nickname;
    this.authorities = List.of(new SimpleGrantedAuthority("ROLE_USER"));
  }

  public Long getId() { return id; }

  @Override public String getUsername() { return nickname; }
  public String getEmail() { return email; }
  @Override public Collection<? extends GrantedAuthority> getAuthorities() { return authorities; }

  @Override
  public String getPassword() {
    return password;
  }

  @Override public boolean isAccountNonExpired()     { return true; }
  @Override public boolean isAccountNonLocked()      { return true; }
  @Override public boolean isCredentialsNonExpired() { return true; }
  @Override public boolean isEnabled()               { return true; }
}

File: Backend/src/main/kotlin/com/luckyseven/backend/domain/member/service/MemberDetails.kt
package com.luckyseven.backend.domain.member.service.utill

import com.luckyseven.backend.domain.member.entity.Member
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

open class MemberDetails : UserDetails {
    val id: Long
    private val password: String
    private val email: String
    private val nickname: String
    private val authorities: Collection<GrantedAuthority>
    
    constructor(member: Member) {
        this.id = member.id!!
        this.email = member.email
        this.nickname = member.nickname
        this.password = member.password
        this.authorities = listOf(SimpleGrantedAuthority("ROLE_USER"))
    }
    
    constructor(id: Long, password: String, email: String, nickname: String) {
        this.id = id
        this.password = password
        this.email = email
        this.nickname = nickname
        this.authorities = listOf(SimpleGrantedAuthority("ROLE_USER"))
    }
    
    override fun getUsername(): String = email
    

    fun getEmail(): String = email
    
    override fun getAuthorities(): Collection<GrantedAuthority> = authorities
    
    override fun getPassword(): String = password
    
    override fun isAccountNonExpired(): Boolean = true
    
    override fun isAccountNonLocked(): Boolean = true
    
    override fun isCredentialsNonExpired(): Boolean = true
    
    override fun isEnabled(): Boolean = true
} 